### PR TITLE
Goonchat UTF-8 support and better string highlighting.

### PR DESCRIPTION
**Pull request by PJB3005**
# UTF-8

Also Russians now say сука блять

http://i.imgur.com/OTFsooS.png

How it works:

BYOND is Windows-1252, but because of how text works, if it reads a file or string in a code file it can't actually read the UTF-8 correctly, and it basically produces garble in Windows-1252.
But because it sends this garble to interfaces literally, if you specify the charset in the HTML to be UTF-8, IE parses it as UTF-8 successfully.

Issue with goonchat is that to send the actual data it needs to be `url_encode()`'d, and BYOND internally does the decoding manually so we trick it by encoding it twice, and then making the Javascript decode it.
# String highlighting.

Goonchat now uses mark.js to do string highlighting.

Credits to @Krausus, ParadiseSS13/Paradise#5450

### CODE REVIEW

File: goon/code/datums/browserOutput.dm [hunk 1]
```diff
@@ -257,4 +257,5 @@ For the main html chat area
 
 		message = replacetext(message, "\n", "<br>")
 
-		target << output(url_encode(message), "browseroutput:output")
+		// url_encode it TWICE, this way any UTF-8 characters are able to be decoded by the Javascript.
+		target << output(url_encode(url_encode(message)), "browseroutput:output")
```
Comment (ComicIronic) [inline goon/code/datums/browserOutput.dm:262]: Gee Bill, double encode?

Comment (PJB3005) [inline goon/code/datums/browserOutput.dm:262]: Did you even read the comment?

Comment (Exxion) [inline goon/code/datums/browserOutput.dm:262]: You didn't even get the meme right

**Comment by clusterfack** [comment]
:+1:

**Comment by ghost** [comment]
сука блять/10
картошка/10
водка/10
коммунизм/10

**Comment by Iamgoofball** [comment]
https://github.com/PJB3005/vgstation13/pull/2

**Comment by PJB3005** [comment]
Gonna remove NFD because I added something which may or may not need review.

**Comment by Coureere** [comment]
балалайка ещё

**Comment by GeneralGoose** [comment]
а буква ё работает или нихуя?

**Comment by clusterfack** [comment]
cyka blyat

**Pull request merged**